llvm/test/tools/llvm-profdata/sample-split-layout.test

```diff
@@ -0,0 +1,2 @@
+RUN: llvm-profdata merge --sample --extbinary --split_layout %p/Inputs/sample-profile.proftext -o %t-output
+RUN: diff %t-output %p/Inputs/split-layout.profdata
```

Comment:
Is there a way to test this without checking in the binary file `split-layout.profdata`?

If you must use a binary file, consider using `printf` to create the file so changes can be reviewed.
https://github.com/llvm/llvm-project/blob/74125fa5ef3e730e53a2685ad7d930a8c685fff9/llvm/test/tools/llvm-cgdata/error.test#L25-L29

Reply:
One way to avoid checking in binary file is to convert it to text format (with `llvm-profdata merge --text`) as input, and convert it back to binary file with a `RUN` line.

Reply:
Yes, that sounds good.

Reply:
Note that there is no way to represent flat profile in text mode, I will have to implement that

Reply:
And there's no point to represent that in text mode, because the point of split layout is to allow sample profile reader to skip reading an entire section when used by ThinLTO since samples without inline callsites provide no info to advise inlining. In text mode the profile is read sequentially anyways so you can't skip anything

Reply:
> Note that there is no way to represent flat profile in text mode, I will have to implement that

Maybe we can try `--show-sec-info-only` to check the section size. The size would change if others remove what you adds.

Reply:
Checking section info only sounds good to me.